doc/asab/storage.rst

```diff
@@ -0,0 +1,309 @@
+.. currentmodule:: asab.storage
+
+Storage
+=======
+
+ASAB provides a service for storing data. Data can be stored in memory or in dedicated document database. For now, `MongoDB <https://www.mongodb.com/>`_ and `Elastic Search <https://www.elastic.co/>`_ databases are supported.
+
+Specification of the storage type
+---------------------------------
+
+In order to use `asab.storage`, first you have to specify the type of storage. You can add configurations in the config file:
+
+.. code:: ini
+
+    [asab:storage]
+    type=mongodb
+
+or you can set it manually in the ASAB app:
+
+.. code:: python
+
+    import asab
+    import asab.storage
+
+    asab.Config.add_defaults(
+        {
+            'asab:storage': {
+                'type': 'mongodb'
+            }
+        }
+    )
+
+The options for the storage type are:
+
+- `inmemory`: Collects data directly in memory
+- `mongodb`: Collects data using MongoDB database. Depends on `pymongo <https://pymongo.readthedocs.io/en/stable/>`_ and `motor <https://motor.readthedocs.io/en/stable/api-asyncio/asyncio_motor_collection.html>`_ libraries.
+- `elasticsearch`: Collects data using Elastic Search database. Depends on `aiohttp <https://docs.aiohttp.org/en/latest/>`_ library.
+
+Although these three databases are different, accessing the database and manipulation with collections is done by using the same methods.
+
+For accessing the storage, simply `add asab.storage.Module`` when initializing and register the service.
+
+.. code:: python
+
+    class MyApplication(asab.Application):
+
+        async def initialize(self):
+
+            self.add_module(asab.storage.Module)
+
+        async def main(self):
+            storage = self.get_service("asab.StorageService")
+
+
+Manipulation with databases
+---------------------------
+
+Upsertor
+~~~~~~~~
+
+Upsertors are used for manipulations with databases. Upsertor is an object that works like a pointer to the specified database and optionally to object id.
```

Comment:
The whole Storage Service is used to manipulate with database. First sentence says nothing new.

![image](https://user-images.githubusercontent.com/82450969/226674826-82d606b6-2c22-42bf-8f47-ae715ba552d6.png)
I know it is not 100% true for this implementation, but it is one clear sentence on what "upsert" mean